src/frontends/tensorflow_common/src/op/tobool.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/select.hpp"
+
+using namespace std;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+OutputVector translate_tobool_op(const NodeContext& node) {
+    // (rank(x) == 0 && x != 0) || (rank > 0 && ReduceProd(ShapeOf(x))) > 0
+
+    default_op_checks(node, 1, {"ToBool"});
+    auto x = node.get_input(0);
+
+    // prepare auxiliary zero and one constants of the same type as the inputs
+    auto zero = create_same_type_const_scalar<int32_t>(x, 0);
```

Comment:
@rsato10, please create two `zero` Constants of `x` type and `int32` and use appropriate constant for comparison below.

Reply:
Sorry for the trouble but do you mean
auto zero = create_same_type_const_scalar<int32_t>(x, 0);
auto zero_2 = make_shared<v0::Constant>(element::i32, x, 0);

Reply:
@rsato10, please fix build. You can observe build logs in `openvino-lin` job

Reply:
../../openvino/src/frontends/tensorflow_common/src/op/tobool.cpp:27:19: note: in instantiation of function template specialization 'std::make_shared<ov::op::v0::Constant, const ov::element::Type &, ov::Output<ov::Node> &, int>' requested here
    auto zero_2 = make_shared<v0::Constant>(element::i32, x, 0);
                  ^
../../openvino/src/core/include/openvino/op/constant.hpp:85:5: note: candidate constructor template not viable: no known conversion from 'ov::Output<ov::Node>' to 'const ov::Shape' for 2nd argument
    Constant(const element::Type& type, const Shape& shape, T value) : Constant(false, type, shape) {
 Does this mean auto_zero2 is not correct?

Reply:
Constant constructor requires a shape by the second argument:

```
auto zero_2 = make_shared<v0::Constant>(element::i32, Shape{}, 0);
```

Reply:
@rsato10, I fixed it:)